pkg/kubectl/resource/mapper.go

```diff
@@ -50,8 +52,16 @@ type Mapper struct {
 func (m *Mapper) InfoForData(data []byte, source string) (*Info, error) {
 	versions := &runtime.VersionedObjects{}
 	_, gvk, err := m.Decode(data, nil, versions)
+	var obj runtime.Object
+	var versioned runtime.Object
+	if registered.IsThirdPartyAPIGroupVersion(gvk.GroupVersion()) {
+		obj, err = runtime.Decode(thirdpartyresourcedata.NewCodec(nil, gvk.Kind), data)
+		versioned = obj
```

Comment:
I'm not sure if this is correct. Have you tried `kubectl apply` on a thirdparty object?